internal/helper_test.go

```diff
@@ -0,0 +1,43 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package internal
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestUnsafeByteSlice2String(t *testing.T) {
+	tests := [][]byte{
+		[]byte("hello world"),
+		[]byte(`{"some":"json", "simple": true}`),
+		[]byte(`multi
+		line
+		test`),
+	}
+
+	for _, test := range tests {
+		test := test
```

Comment:
This is not needed since the closure completes before the next loop iteration.

Reply:
Is there a functional difference that makes it important to remove it?
In the standart libray, we can find both:
With (similar to here):
https://sourcegraph.com/github.com/golang/go/-/blob/src/cmd/vet/vet_test.go?L104-105
Without (similar to suggestion):
https://sourcegraph.com/github.com/golang/go/-/blob/src/internal/reflectlite/all_test.go?L349:3

Is this a blocker?

Reply:
The difference between those two cases is that the first (using the new decl) runs the tests in parallel and so does not satisfy the condition "the closure completes before the next loop iteration".